patches/src/main/kotlin/app/revanced/patches/instagram/misc/signature/Fingerprints.kt

```diff
@@ -0,0 +1,15 @@
+package app.revanced.patches.instagram.misc.signature
+
+import app.revanced.patcher.fingerprint
+import com.android.tools.smali.dexlib2.AccessFlags
+import com.android.tools.smali.dexlib2.Opcode
+
+internal val isValidSignatureClassFingerprint = fingerprint {
+    strings("The provider for uri '", "' is not trusted: ")
+}
+
+internal val isValidSignatureMethodFingerprint = fingerprint {
+    accessFlags(AccessFlags.PUBLIC, AccessFlags.FINAL)
+    parameters("L", "Z")
+    returns("Z")
+}
```

Comment:
```suggestion
internal val isValidSignatureMethodFingerprint = fingerprint {
    parameters("L", "Z")
    returns("Z")
    custom { method, _ -> method.indexOfFirstInstruction {
            getReference<MethodReference>()?.name == "keySet"
        } >= 0
    }
}
```

Should probably make the fingerprint a bit more specific, and can stop relying on access flags too since they dont really change much

Reply:
This fingerprint is used to match on a single class with only a handful of methods, so I don't think it needs any additional specificity?

Reply:
I think it's worth it, the class has a lot of methods with parameters and return which are almost the same

Reply:
I can't really push to the branch so the max I can do is suggest another change

Reply:
I'll handle it